### PR TITLE
fix: preserve date values set to `null`

### PR DESCRIPTION
**Pull request by bajtos**
Before this change, when the client sent `end_at: null`, we would convert that value to `Date(null)`, which is the Unix epoch.

After this change, we preserve null values as null.

Links:
- https://github.com/filecoin-station/spark/issues/43

### CODE REVIEW

File: test/test.js [hunk 1]
```diff
@@ -213,6 +213,43 @@ describe('Routes', () => {
...
+
+      const { rows: [measurementRow] } = await client.query(`
+          SELECT *
+          FROM measurements
+          WHERE id = $1
+        `, [
+        id
+      ])
+
+      assert.strictEqual(measurementRow.status_code, null)
```
Comment (bajtos) [inline test/test.js:249]: `status_code` is NULLABLE in the database. When the measurement does not specify the status code at all (value is `undefined`), the roundtrip through the database converts the value to `null`.

This PR does not change the existing behaviour, just adds a test to document it.

🤷🏻‍♂️

File: index.js [hunk 2]
```diff
@@ -285,3 +285,8 @@ export const createHandler = async ({
       })
   }
 }
+
+const parseDateStringOrNull = (strOrNull) => {
+  if (strOrNull === undefined || strOrNull === null) return undefined
```
Comment (juliangruber) [inline index.js:290]: slightly confusing: It's called `OrNull` but returns `undefined` and accepts `null` or `undefined`

Comment (juliangruber) [reply index.js:290]: What about `parseDateStringOrUndefined`?

Comment (bajtos) [reply index.js:290]: I decided to use `parseOptionalDate`

**Review by juliangruber — APPROVED** [review]

**Pull request merged**